rest/client/rest.go

```diff
@@ -20,6 +20,29 @@ import (
 	"github.com/pkg/errors"
 )
 
+// respErrorf attempts to read a gimlet.ErrorResponse from the response body
+// JSON. If successful, it returns the gimlet.ErrorResponse wrapped with the
+// HTTP status code and the formatted error message. Otherwise, it returns an
+// error message with the HTTP status and raw response body.
+func respErrorf(resp *http.Response, format string, args ...interface{}) error {
+	wrapError := func(err error) error {
```

Comment:
It might be cleaner to have this as a separate function rather than an anonymous closure in here. I don't feel strongly about this though

Reply:
I think it's probably okay to leave this as-is since it's only used in this one function and won't really mess up the stack trace.